### PR TITLE
Provide a Flatpak manifest

### PR DESCRIPTION
**Pull request by meisenzahl**
Closes https://github.com/elementary/feedback/issues/33

### CODE REVIEW

**Comment by meisenzahl** [comment]
I could use some help.

Feedback starts now, but the submenus for *Default Apps*, *Panel Indicators* and *System Settings* are empty. I assume that they are filled dynamically and that starting it as Flatpak will not find the applications.

File: io.elementary.feedback.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+finish-args:
+  - '--share=ipc'
+  - '--share=network'
+  - '--socket=fallback-x11'
+  - '--socket=wayland'
+
+  # needed for perfers-color-scheme
+  - '--system-talk-name=org.freedesktop.Accounts'
+
+  - '--metadata=X-DConf=migrate-path=/io/elementary/feedback/'
```
Comment (Marukesu) [inline io.elementary.feedback.yml:17]:
```suggestion
  - '--metadata=X-DConf=migrate-path=/io/elementary/feedback/'
cleanup:
  - '/include'
  - '/lib/pkgconfig'
  - '/lib/girepository-1.0'
  - '/share/gir-1.0'
  - '/share/man'
  - '/share/vala'
  - '*.la'
  - '*.a'
```

File: io.elementary.feedback.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+modules:
+  - name: appstream
+    buildsystem: meson
+    config-opts:
+      - '-Dvapi=true'
```
Comment (Marukesu) [inline io.elementary.feedback.yml:22]:
```suggestion
      - '-Dvapi=true'
      - '-Dapidocs=false'
      - '-Dinstall-docs=false'
```

File: io.elementary.feedback.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    sources:
+      - type: git
+        url: https://github.com/ximion/appstream
+        tag: 'v0.12.11'
+    modules:
+      - name: xapian
```
Comment (Marukesu) [inline io.elementary.feedback.yml:28]:
```suggestion
      - name: xapian
        cleanup:
          - '/lib/cmake'
          - '/share/alocal'
          - '/share/doc'
```

File: io.elementary.feedback.yml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        sources:
+          - type: archive
+            url: https://oligarchy.co.uk/xapian/1.4.17/xapian-core-1.4.17.tar.xz
+            md5: 64cd582a4495c5cbda8e06e63f8ce4f4
+      - name: yaml
+        sources:
+          - type: git
+            url: https://github.com/yaml/libyaml
+            tag: '0.2.5'
+      - name: Intltool
```
Comment (Marukesu) [inline io.elementary.feedback.yml:38]:
```suggestion
      - name: Intltool
        cleanup:
          - '*'
```

File: io.elementary.feedback.yml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        sources:
+          - type: archive
+            url: https://launchpad.net/intltool/trunk/0.51.0/+download/intltool-0.51.0.tar.gz
+            md5: 12e517cac2b57a0121cda351570f1e63
+      - name: xmlto
+        buildsystem: autotools
```
Comment (Marukesu) [inline io.elementary.feedback.yml:44]: autotools is the default buildsystem, no need to set it here.

File: io.elementary.feedback.yml [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        sources:
+          - type: archive
+            url: https://releases.pagure.org/xmlto/xmlto-0.0.28.tar.bz2
+            md5: 93bab48d446c826399d130d959fe676f
+      - name: lmdb
+        buildsystem: simple
+        build-commands:
+          - 'make'
+          - 'make install prefix=/app'
```
Comment (Marukesu) [inline io.elementary.feedback.yml:53]:
```suggestion
        no-autogen: true
        make-install-args:
          - 'prefix=/app'
```

No need to use simple here.

File: io.elementary.feedback.yml [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        subdir: 'libraries/liblmdb'
+        sources:
+          - type: git
+            url: https://git.openldap.org/openldap/openldap
+            tag: 'LMDB_0.9.27'
+      - name: stemmer
+        buildsystem: cmake
```
Comment (Marukesu) [inline io.elementary.feedback.yml:60]:
```suggestion
        buildsystem: cmake-ninja
```

**Review by Marukesu — COMMENTED** [review]
Need some cleanups definitions and others minors changes.

**Comment by Marukesu** [comment]
about the appstream error, the flatpak don't have access to the host metadata, so we need to include the needed files with the flatpak, or place them in a location that the flatpak can read if setted with `appstream_pool.add_metadata_location`.

it fix the __Painel Indicators__ and __System Settings__, but for __Default Appliications__ it fails in create the `DesktopAppInfo`.

**Comment by meisenzahl** [comment]
@Marukesu thanks for your feedback!

@danrabbit what is your opinion how we want to continue?

**Comment by danrabbit** [comment]
I think it would definitely be preferable to poke the right holes in the sandbox so it can read the installed apps from the system. It would be ultimately more maintainable to reflect what the user actually has installed

**Comment by Marukesu** [comment]
adding `appstream_pool.add_metadata_location ("/run/host/usr/share/metainfo")` to MainWindow.vala makes the native appstream readable in the flatpak (running with `--filesystem=host:ro`), but it can't read the .desktop files.

```
(io.elementary.feedback:2): GLib-GIO-CRITICAL **: 18:52:32.607: g_app_info_get_display_name: assertion 'G_IS_APP_INFO (appinfo)' failed

(io.elementary.feedback:2): GLib-GIO-CRITICAL **: 18:52:32.607: g_app_info_get_icon: assertion 'G_IS_APP_INFO (appinfo)' failed

** (io.elementary.feedback:2): CRITICAL **: 18:52:32.607: feedback_repo_row_construct: assertion 'title != NULL' failed

(io.elementary.feedback:2): GLib-GObject-CRITICAL **: 18:52:32.607: g_object_ref_sink: assertion 'G_IS_OBJECT (object)' failed

(io.elementary.feedback:2): Gtk-CRITICAL **: 18:52:32.607: gtk_container_add: assertion 'GTK_IS_WIDGET (widget)' failed
```

**Comment by Marukesu** [comment]
Well, since `DesktopAppInfo` won't work inside the flatpak, i changed the app to use only the appstream metadata and worked. But, to have the icons, the appdata need the `icon` tag (none of the needed apps have it, through).

Captura de tela de 2020-11-26 22-35-20

**Review by danrabbit — APPROVED** [review]
This at least builds and shows apps, I think we can follow up later with a solution for all app icons. That probably means adding icons to other components appdata files. So I think this is good to merge :)

**Pull request merged**